engine/crates/integration-tests/tests/postgres/find_one/joins.rs

```diff
@@ -1430,3 +1430,65 @@ fn one_to_one_with_one_to_many_joins_parent_side() {
 
     expected.assert_eq(&response);
 }
+
+#[test]
+fn two_foreign_keys_to_same_table() {
+    let response = query_postgres(|api| async move {
+        let setup = [
+            r#"CREATE TABLE public.colors (
+                id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+                name VARCHAR(255) NOT NULL UNIQUE,
+                rgb INT NOT NULL
+            );"#,
+            r#"CREATE TABLE public.users (
+                id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+                name VARCHAR(255) NOT NULL UNIQUE,
+                favorite_color_id INT NOT NULL REFERENCES colors(id),
+                least_favorite_color_id INT REFERENCES colors(id)
+            );"#,
+            r#"INSERT INTO public.colors (name, rgb) VALUES 
+              ('rebeccapurple',  0x663399),
+              ('tomato',         0xff6347),
+              ('skyblue',        0x87ceeb),
+              ('maroon',         0x800000);"#,
+            r#"INSERT INTO public.users (name, favorite_color_id, least_favorite_color_id) VALUES
+              ('Guignol', 1, 2),
+              ('Gnafron', 2, NULL),
+              ('Flageolet', 3, 2),
+              ('Canezou', 3, 4);
+             "#,
+        ];
+
+        for statement in setup {
+            api.execute_sql(statement).await;
+        }
+
+        std::thread::sleep(std::time::Duration::from_millis(1000));
+
+        let query = indoc! {r#"
+            query {
+              gnafron: users(by: { name: "Gnafron" }) {
+                name
+                colorsByFavoriteColorId { id name rgb }
+                colorsByLeastFavoriteColorId { id name rgb }
+              }
+              guignol: users(by: { name: "Guignol" }) {
+                name
+                colorsByFavoriteColorId { id name rgb }
+                colorsByLeastFavoriteColorId { id name rgb }
+              }
+              colors(by: { name: "tomato" }) {
+                  id
+                  usersByFavoriteColorId(first: 10) { edges { node { id name favoriteColorId leastFavoriteColorId } } }
+                  usersByLeastFavoriteColorId(first: 10) { edges { node { id name favoriteColorId leastFavoriteColorId } } }
+              }
+            }
+        "#};
+
+        api.execute(query).await
+    });
+
+    let expected = expect![[""]];
```

Comment:
Should we have a test that returns something too?

Reply:
PR is not submitted for review yet, I haven't updated the snapshot